lib/api/concrexit_api_repository.dart

```diff
@@ -121,6 +121,26 @@ class ConcrexitApiRepository implements ApiRepository {
     }
   }
 
+  /// A wrapper for making requests that handles logging the requests in debug mode.
+  Future<Response> _logRequest(
+      Uri url, Future<Response> Function() request) async {
+    try {
+      var response = await _handleExceptions(request);
+
+      if (kDebugMode) {
+        print('url: $url, response code: ${response.statusCode}');
+      }
+
+      return response;
+    } on ApiException catch (exception) {
+      if (kDebugMode) {
+        print('url: $url, exception: ${exception.message}');
+      }
+
+      rethrow;
+    }
+  }
+
```

Comment:
Maybe a cleaner solution would be to extend the `oauth2.Client`? that way there is no need to add this function to every request